src/vs/workbench/api/browser/mainThreadEditors.ts

```diff
@@ -139,7 +139,13 @@ export class MainThreadTextEditors implements MainThreadTextEditorsShape {
 		if (!editor) {
 			return undefined;
 		}
-		return this._editorLocator.findTextEditorIdFor(editor);
+		// Composite editors are made up of many editors so we return the active one at the time of opening
+		const editorControl = editor.getControl();
```

Comment:
@bpasero This was partially taken from https://github.com/microsoft/vscode/blob/79c10fac4789a437c15e4d0574723fe2649f0ef2/src/vs/workbench/services/editor/browser/codeEditorService.ts#L32-L48 I wonder if we could benefit from exposing a static `getCodeEditorFromControl(editor: IEditor | IDiffEditor | undefined)` function that we could use to get a code editor. Would it then be safe to always use `getIdOfCodeEditor` instead of `findTextEditorIdFor(editor)`.

Reply:
There already is something like that for code editors:

https://github.com/microsoft/vscode/blob/028186209f6b4db55b9cf10ab7edb26206ef237c/src/vs/editor/browser/editorBrowser.ts#L1221-L1231

I don't mind exposing a new helper method that reuses the code, but I would not change the semantic of the `getActiveCodeEditor` method without knowing the consequences such a change would have.

Reply:
Yeah I wasn't looking to change `getActiveCodeEditor`. My question is it safe to not use `findTextEditorIdFor` in `showTextDocument` and instead get away with always using the code editor

Reply:
I yield the floor for @jrieken to answer

Reply:
Sorry, no time for a deep review but my intuition is the the other ways around. If you don't use `findTextEditorIdFor` etc then you are likely doing it wrong. The `$tryShowTextDocument` must return something that API layer has identified as editor, that must not match the workbench layer. Please don't merge this yet

Reply:
They're both returning an id that's valid within the `this._textEditors` array so I think it's all ok, but I'll hold off for your review 👍

Reply:
You are right @lramos15. This looks correct and I tested that it works. Maybe consider to move the knowledge of composite editors directly into the [findTextEditorIdFor](https://github.com/microsoft/vscode/blob/aca007480559ff91fcecd31cafe01955e433b379/src/vs/workbench/api/browser/mainThreadDocumentsAndEditors.ts#L412)-implementation.

Reply:
Well as @bpasero pointed out https://github.com/microsoft/vscode/pull/158477#discussion_r949093073 there is a way to get the code editor of an `editorPane`. I think we could enrich that with composite editors and then just call `return getIdOfCodeEditor(getCodeEditor(editor));`and stop using `findTextEditorIdFor` even with basic text editors